src/server/worker/master.go

```diff
@@ -691,7 +693,7 @@ func (a *APIServer) runJob(ctx context.Context, jobInfo *pps.JobInfo, pool *pool
 				}
 				jobInfo.Finished = now()
 				jobInfo.StatsCommit = statsCommit
-				return a.updateJobState(stm, jobInfo, pps.JobState_JOB_FAILURE)
+				return a.updateJobState(stm, jobInfo, pps.JobState_JOB_FAILURE, fmt.Sprintf("failed to process datum: %v", failedDatum))
```

Comment:
This is likely to result in a pretty large reason message, why not just do the datum ID here and then they can get more info about it by doing `inspect-datum`